### ISSUE TITLE
Download url erro

### ISSUE THREAD

**Issue by hhuseyn** [Athlon1600/youtube-downloader#55]
Hi.i run this project but some download url is not work
example  youtube id "tzi6tkCbHBc";
get download ur: "https://r3---sn-aigl6ney.googlevideo.com/videoplayback?expire=1589764766&ei=PY7BXr_iOOaOxN8P4_uK2Ao&ip=64.227.40.195&id=o-AMBaUnaG5QHVgity_wOr56dfF-0Y2KMxzq0Aoc9dEY82&itag=18&source=youtube&requiressl=yes&mh=pI&mm=31%2C29&mn=sn-aigl6ney%2Csn-aigzrney&ms=au%2Crdu&mv=m&mvi=2&pl=21&gcr=gb&initcwndbps=162500&vprv=1&mime=video%2Fmp4&gir=yes&clen=12803988&ratebypass=yes&dur=184.575&lmt=1587863251161273&mt=1589743056&fvip=6&c=WEB&txp=5431432&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cgcr%2Cvprv%2Cmime%2Cgir%2Cclen%2Cratebypass%2Cdur%2Clmt&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIhAN2ZdmVBWsUFjlC_M7mE55sZ_L-zLD2_jRAdf564M-eFAiBvFkh7A4_WALs3gds1R5nZQtrQkTbYCpZePIg1gLa-iA%3D%3D&sig=AOq0QJ8wRQIgHdg1J2s2rZ6Mvvw9BYouvls_hUOCBRNq5kKdfMjY22QCIQCx1xgFfRLeZsbQAUdG4D0M0gvEMF2-C2tvIc1K4VNrPA=="

how can fix this problem?
php version :7.1.33
composer version:1.10.6

**Comment by Athlon1600** [comment]
If I understand this correctly: video download link that was generated on your localhost works fine locally, but gives you a 403 forbidden error (or something similar) when trying to access that same link from a different server?  
I can't seem to replicate this from my end...

**Comment by tienanhbui** [comment]
Same to me!
I am tested with many videos. Some video has copyright and fullhd video quality has higher probability return 403 code.
I am reading function `decode`  in `SignatureDecoder` class. Can you explain about this function? Maybe this function not working.
Thanks!

**Comment by vnddr** [comment]
Hello! I faced quite the same problem as the topicstarter. Video download link extraction works fine, but the link returns 403.
I thought that this problem occurs because i'm using proxies to get the download links, but I've tested it without proxies from both remote server and my local pc and the result is still the same 403 error.
Unfortunately, I haven't still found any obvious signs about the videos which can cause the problem.
Maybe this helps: this issue always reproduces for me with these videos:
cgdpUot6sW8
E3sXO-KJpZM

**Comment by Athlon1600** [comment]
Yup. There was an issue with `SignatureDecoder` class. Something has changed within youtube recently that broke the logic. `youtube-dl` that I have installed on my computer stopped working too, until I updated it...

I just put in a fix, so just get the new v2.0.9: 
https://github.com/Athlon1600/youtube-downloader/releases

and it should be working now!

**Comment by ubaidsuhail** [comment]
It should still given error on live server

**Comment by ubaidsuhail** [comment]
It is working fine on local host but gives error on live server.

**Comment by rajatvermaiam** [comment]
yeah it is still gving error..
